crates/sui-framework/src/lib.rs

```diff
@@ -132,6 +137,58 @@ pub fn run_move_unit_tests(
     )
 }
 
+pub fn build_move_package(
+    path: &Path,
+    build_config: BuildConfig,
+) -> SuiResult<Vec<CompiledModule>> {
+    let pkg = build_move_package_with_deps(path, build_config)?;
+    verify_framework_version(&pkg)?;
+    filter_package_modules(&pkg)
+}
+
+/// Version of the framework code that the binary used for compilation expects should be the same as
+/// version of the framework code bundled as compiled package's dependency and this function
+/// verifies this.
+fn verify_framework_version(pkg: &CompiledPackage) -> SuiResult<()> {
+    // We stash compiled modules in the Modules map which is sorted so that we can compare sets of
+    // compiled modules directly.
+
+    let dep_framework_modules = pkg.all_modules_map().iter_modules_owned();
+    let dep_framework: Vec<&CompiledModule> = dep_framework_modules
+        .iter()
+        .filter(|m| *m.self_id().address() == SUI_FRAMEWORK_ADDRESS)
+        .collect();
+
+    let framework_modules = Modules::new(get_sui_framework().iter()).iter_modules_owned();
```

Comment:
What does `get_sui_framework` do?

Reply:
It forces lazy evaluation of a static:
https://github.com/MystenLabs/sui/blob/3c6e934e387c17ad86d6d0793773a7cb05ff041d/crates/sui-framework/src/lib.rs#L46

This static grabs a serialized version of the framework packages crated in the pre-build [script](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/build.rs) and deserializes it to a vector of `CompiledModules`.